### PR TITLE
Pop input widget

### PR DESCRIPTION
**Pull request by robinr**
The PopInput_Widget contains all the features of the defined PopInput Widget
a) basic-input
b) input-disallow
c) input-no-title-label
d) input-spellcheck 
e) input-validation-rules
f) input-width-200
g) input-width-300
h) x-y-offset

NOTE : The testcafe is not able to capture the image of the squiggle redline, functionally the feature of spellcheck works
             on the VRT unable to validate via image comparison.

### CODE REVIEW

File: components/playback/controls/player/PopInput.vue [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+Vue.component("BFormInput", BFormInput);
+Vue.component("BButton", BButton);
+import "bootstrap/dist/css/bootstrap.min.css";
+Vue.use(uuid);
+
+export default {
+  name: "PopInput",
+  props: {
+    title_label: { type: String, default: "", required: false }, // title_text (str) (optional, defaults to empty string "")
+    key_placeholder: { type: String, default: "" }, // placeholder (str)
```
Comment (eli88fine) [inline components/playback/controls/player/PopInput.vue:86]: is there a reason you want to make up a new name `key_placeholder` for this? can we just use `placeholder` so it has a similar and familiar syntax as the disabled used for the input DOM element

File: components/playback/controls/player/PopInput.vue [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    invalid_text: { type: String, default: "" }, // invalid_text (str)
+    input_check: { type: Boolean, default: true, required: false }, // spellcheck (optional bool=True)
+    user_key: { type: String, default: "", required: false }, // field_value (str) (optional, defaults to empty string "")
```
Comment (eli88fine) [inline components/playback/controls/player/PopInput.vue:89]: is there a reason you want to make up a new name `user_key` for this? can we just use `value` so it has a similar and familiar syntax as the disabled used for the input DOM element

File: components/playback/controls/player/PopInput.vue [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+import "bootstrap/dist/css/bootstrap.min.css";
+Vue.use(uuid);
+
+export default {
+  name: "PopInput",
+  props: {
+    title_label: { type: String, default: "", required: false }, // title_text (str) (optional, defaults to empty string "")
+    key_placeholder: { type: String, default: "" }, // placeholder (str)
+    invalid_text: { type: String, default: "" }, // invalid_text (str)
+    input_check: { type: Boolean, default: true, required: false }, // spellcheck (optional bool=True)
```
Comment (eli88fine) [inline components/playback/controls/player/PopInput.vue:88]: is there a reason you want to make up a new name `input_check` for this? can we just use `spellcheck` so it has a similar and familiar syntax as the disabled used for the input DOM element

File: components/playback/controls/player/PopInput.vue [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+</script>
+<style type="text/css">
+.div__popinput-background {
+  transform: rotate(0deg);
+  box-sizing: border-box;
+  padding: 0px;
+  margin: 0px;
+  background: rgb(17, 17, 17);
+  position: absolute;
+  /* height: 100px; */
```
Comment (eli88fine) [inline components/playback/controls/player/PopInput.vue:129]: remove old commented code when submitting pull requests, unless there is a specific reason to leave it in

File: components/playback/controls/player/PopInput.vue [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+  background-color: #2f2f2f;
+}
+
+.div__popinput-controls-content-input-widget {
+  pointer-events: all;
+  transform: rotate(0deg);
+  overflow: hidden;
+  position: absolute;
+  height: 45px;
+  /* top: 40px; */
```
Comment (eli88fine) [inline components/playback/controls/player/PopInput.vue:191]: remove old commented code when submitting pull requests, unless there is a specific reason to leave it in

File: components/playback/controls/player/PopInput.vue [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+.div__popinput-controls-content-input-feedback {
+  line-height: 1;
+  transform: rotate(0deg);
+  padding: 0px;
+  margin: 0px;
+  overflow-wrap: break-word;
+  color: rgb(229, 74, 74);
+  font-family: Muli;
+  position: absolute;
+  /* top: 88px; */
```
Comment (eli88fine) [inline components/playback/controls/player/PopInput.vue:219]: remove old commented code when submitting pull requests, unless there is a specific reason to leave it in

File: pages/playback/controls/player/popup-input/basic-input.vue [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+      :invalid_text="error_text"
+      :user_key.sync="entrykey"
+      :input_width="entry_width"
+      :block="disallow_entry"
+    ></PopInput>
+  </div>
+</template>
+
+<script>
+import { PopInput } from "../../../../../dist/mantarray.common";
```
Comment (eli88fine) [inline pages/playback/controls/player/popup-input/basic-input.vue:16]: you can use `from "@/dist/mantarray.common"` similar to `@/components` .  That way if this file gets moved to reorganize the folder structure, it will still correctly point to the `dist` folder without having to modify the file path

File: pages/playback/controls/player/popup-input/input-disallow.vue [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+    PopInput,
+  },
+  data: function () {
+    return {
+      entrykey: "",
+    };
+  },
+  watch: {
+    entrykey() {
+      // console.log(this.entrykey);
```
Comment (eli88fine) [inline pages/playback/controls/player/popup-input/input-disallow.vue:34]: is all this code needed here?  if input is disabled is there any need for all this validation logic?  test cases (and test pages) should include just the minimum amount of code needed to run the test, that way it is easy for someone to read the test and understand what is happening

File: tests/unit/components/playback/controls/PopInput.spec.js [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+      propsData,
+      store,
+      localVue,
+    });
+    const target_div = wrapper.find(
+      ".div__popinput-controls-content-input-feedback"
+    );
+    expect(target_div.text()).toStrictEqual("This field is required");
+  });
+  test("When the PopInput is mounted, Then the user enters few charters in the input, verify that the spellcheck is set to false", async () => {
```
Comment (eli88fine) [inline tests/unit/components/playback/controls/PopInput.spec.js:97]: is there a reason you have the user type some characters in this test? it doesn't seem like it would be necessary to test whether the spellcheck is set to false, but maybe I'm missing something

**Review by eli88fine — CHANGES_REQUESTED** [review]
Much improved from last time, thanks for creating the new features and tests.  Just a few minor items to address before the merge

File: tests/unit/components/playback/controls/PopInput.spec.js (hunk 9 above)
Comment (robinr) [reply tests/unit/components/playback/controls/PopInput.spec.js:97]: The Eslint rules ensures the "Then the" words in the test title, so to pass Eslint added the text and for any user to verify from an npm run dev the validation of spellcheck functional work, have the description 'enters few characters in the input'.  The second point PopInput itself could be modified to pass an improper prop instead of :spellcheck it can be modified to :s_pellcheck of some sort, in order to prevent such issue, included the check for the attribute 'spellcheck=false' which is sent on the prop 'input_check: false'.

**Comment by robinr** [comment]
Rest of all the comments accepted will do the changes.
